scripts/linters/pylint_extensions.py

```diff
@@ -1746,6 +1756,29 @@ def _check_punctuation(self, line, line_num):
                 excluded_phrase_at_beginning_of_line)):
             self.add_message('invalid-punctuation-used', line=line_num)
 
+    def _check_trailing_comment(self, line, line_num):
+        """Checks if the comment is an invalid trailing comment or an acceptable
+        pylint pragma.
+
+        Args:
+            line: str. The current line of comment.
+            line_num: int. Line number of the current comment.
+        """
+        # Extract comment from line.
+        split_line = line.split('#')
+        trail_comment = split_line[1]
+        excluded_phrase_at_beginning_of_line = any(
+            trail_comment[1:].startswith(word)
+            for word in EXCLUDED_PHRASES) or any(
+                trail_comment.startswith(word) for word in EXCLUDED_PHRASES)
+        # Check for pylint pragma.
+        if excluded_phrase_at_beginning_of_line:
+            if not trail_comment[0].isspace():
+                self.add_message('no-space-before-pylint-pragma', line=line_num)
```

Comment:
I think this check can be done at a bit earlier in the function, maybe as the first step (by reusing `_check_space_at_beginning_of_comments` with proper arguments) as it would simplify the code.

Reply:
I wasn't sure how to get the result of `_check_space_at_beginning_of_comments`

Reply:
You can see that we already have an the error msg id  `no-space-at-beginning` ([ref](https://github.com/oppia/oppia/blob/795d2d01ce4cd348bebb814777c5a0a0b717922e/scripts/linters/pylint_extensions.py#L1668-L1670)), so you don't need to create one for yourself. The function isn't returning anything, it adds the error message itself if it finds a comment isn't starting with a space, so you should use that. I hope its usage is clear now, else please let me know.